test/mpi/Makefile_cuda.mtest

```diff
@@ -0,0 +1,32 @@
+##
+## Copyright (C) by Argonne National Laboratory
+##     See COPYRIGHT in top-level directory
+##
+
+## This is an automake makefile fragment that should be included by:
+##
+##   include $(top_srcdir)/Makefile_cuda.mtest
+##
+## This makefile is for CUDA tests (compiled with nvcc).
+
+include $(top_srcdir)/Makefile_single.mtest
+
+AM_DEFAULT_SOURCE_EXT = .cu
+
+# Because libtool does not have a .cu language tag yet, some of the key
+# variables and patter rule is not generated. We manually supply them here.
+
+# Automake silent build macros
+AM_V_CCLD = $(am__v_CCLD_$(V))
+am__v_CCLD_ = $(am__v_CCLD_0)
+am__v_CCLD_0 = @echo "  CCLD    " $@;
+am__v_CCLD_1 =
+
+# the LINK variable
+LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
+       $(LIBTOOLFLAGS) --mode=link $(CXX) $(AM_CFLAGS) $(CFLAGS) \
+       $(AM_LDFLAGS) $(LDFLAGS) -o $@
+
+# How to compile .cu files
+.cu.o:
+	MPICH_CC=$(NVCC) $(CC) -c -o $@ $<
```

Comment:
I get that `nvcc` needs to be invoked, but I'm still confused by the `MPICH_CC=` portion of this rule. What does it do?

Reply:
Oh, I guess we are overriding the compiler in the `mpicc` wrapper. I was only searching the testsuite for clues.

Reply:
Similarly we want to link with `mpicxx` for its mpi baggage and avoid adding -lstdc++. It is all hacky one way or another, but I think this way is more explicit and simpler.

Reply:
Oh, your original question -- `MPICH_CC` is the `mpicc`'s mechanism to use a user-specified compiler other than the one that is used to build mpich.

Reply:
Amended the comment with the last fixup commit.

Reply:
I get it now. I was reading the entire line as a whole instead of recognizing that `MPICH_CC=$(NVCC)` is just an environment setting for the following command invocation `$(CC) -c -o $@ $<`.

Reply:
I was looking for whether `Makefile` support some global environment export, but didn't find one.